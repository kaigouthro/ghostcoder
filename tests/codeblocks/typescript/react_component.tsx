import React, { useState } from 'react';

interface MyComponentProps {
  initialCount?: number;
}

const MyComponent: React.FC<MyComponentProps> = ({ initialCount = 0 }) => {
  const [count, setCount] = useState(initialCount);

  const resetCount = () => {
    setCount(initialCount);
  };

  return (
    <div>
      <p>You clicked {count} times</p>
      <button onClick={() => setCount(count + 1)}>
        Click me
      </button>
    </div>
  );
}

export default MyComponent;
