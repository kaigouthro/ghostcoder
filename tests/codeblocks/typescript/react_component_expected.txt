 0 module ``
  1 import `import React, { useState } from 'react';`
  1 class `interface MyComponentProps`
   2 block_delimiter `{`
   2 code `initialCount?: number`
   2 block_delimiter `;`
   2 block_delimiter `}`
  1 class `const MyComponent: React.FC<MyComponentProps> = ({ initialCount = 0 }) =>`
   2 block_delimiter `{`
   2 code `const [count, setCount]`
    3 code `=`
    3 code `useState`
     4 code `(initialCount)`
    3 block_delimiter `;`
   2 function `const resetCount = () =>`
    3 block_delimiter `{`
    3 code `setCount`
     4 code `(initialCount)`
     4 block_delimiter `;`
    3 block_delimiter `}`
    3 block_delimiter `;`
   2 statement `return`
    3 block_delimiter `(`
    3 code `<div>`
     4 code ``
     4 code `<p>You clicked`
      5 block_delimiter `{`
      5 code `count`
      5 block_delimiter `}`
      5 code `times`
      5 code `</p>`
     4 code ``
     4 code `<button onClick={() => setCount(count + 1)}>`
      5 code `Click me`
      5 code `</button>`
     4 code ``
     4 code `</div>`
    3 block_delimiter `)`
    3 block_delimiter `;`
   2 block_delimiter `}`
  1 code `export default MyComponent;`
  1 space ``
